src/system_identity/client_input.rs

```diff
@@ -0,0 +1,20 @@
+use super::environment::SystemIdentityCreationEnvironment;
+
+/// Represents the input data required to create a System Identity.
+#[derive(Debug, Clone, PartialEq)]
+pub struct SystemIdentityCreationMetadata {
+    pub name: String,
+    pub organization_id: String,
+    pub client_id: String,
+    pub auth_method: AuthMethod,
+    pub environment: SystemIdentityCreationEnvironment,
+}
+
+type ClientSecret = String; // TODO type better
+type PrivateKeyPem = Vec<u8>; // TODO Type better
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum AuthMethod {
+    ClientSecret(ClientSecret),         // L1 method
+    FromLocalPrivateKey(PrivateKeyPem), // L2 method
+}
```

Comment:
I would encapsulate `PrivateKeyPem` into a struct that does not implement debug (or obfuscates it). WDYT?

Reply:
Yes, that was something I wanted to do (hence the comments about typing better xD) but I wanted to have the general architecture and implementation out there so we can start discussing while I make this stricter. Working on it!